File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/persistence/repository/WeekEntityRepository.java
package com.tarbadev.witchcraft.persistence.repository;

import com.tarbadev.witchcraft.persistence.entity.WeekEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface WeekEntityRepository extends JpaRepository<WeekEntity, Integer> {
  Optional<WeekEntity> findByYearAndWeekNumber(Integer year, Integer weekNumber);
}

File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/weeks/persistence/repository/WeekEntityRepository.kt
package com.tarbadev.witchcraft.weeks.persistence.repository

import com.tarbadev.witchcraft.weeks.persistence.entity.WeekEntity
import org.springframework.data.jpa.repository.JpaRepository

import java.util.Optional

interface WeekEntityRepository : JpaRepository<WeekEntity, Int> {
    fun findByYearAndWeekNumber(year: Int?, weekNumber: Int?): Optional<WeekEntity>
}
